### PR TITLE
feat(DORA): Start reporting deploys in production in datadog

### PR DESCRIPTION
**Pull request by george-silva**
Makes the production deploy to report metrics in datadog.

## Description

1. Added a CI step that pushes a deploy metric to datadog. This is useful in
the context of DORA metrics.

## Types of changes

- [ ] Fix (non-breaking change which fixes an issue)
- [ ] Feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Build (Changes that affect the build system or dependencies)
- [ ] Chore (Other changes that do not modify src or test files)
- [ ] CI (Changes to our CI configration and scripts)
- [ ] Docs (Changes to documentation files)
- [ ] Style (Changes to coding style, like linters and prettier)
- [ ] Refactor (Changes that do not add new functionality or alter current ones)
- [ ] Performance (Performance improvements)
- [ ] Test (Adding or changing tests or the testing setup)
- [ ] Revert (Reverting a previous change)

## Checklist:

- [ ] I have made sure my changes work properly on iOS devices running Safari.
- [ ] I have made sure my changes work properly on Android devices running Chrome.
- [ ] I have made sure my changes work properly on Desktop.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by logrock-eng** [comment]
# Netlify Preview Deployed 

 Preview: https://68--logrock-pebbles.netlify.app 

 ## View on Browserstack 

 https://live.browserstack.com/dashboard#os=iOS&os_version=15.0&device_browser=safari&zoom_to_fit=true&full_screen=true&url=https://68--logrock-pebbles.netlify.app&speed=1

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -38,6 +38,18 @@ jobs:
...
+          api-key: ${{ secrets.DATADOG_API_KEY }}
+          metrics: |
+            - type: "count"
+              name: "logrock.deploy"
+              value: 1.0
+              host: ${{ github.repository_owner }}
+              tags:
+                - "project:${{ github.repository }}"
+                - "env:production"
+
```
Comment (lipe-dev) [inline .github/workflows/release.yml]: can you place this after the storybook deploy? deploy of the lib requires npm publish + storybook documentation up to be considered complete.

**Review by lipe-dev — APPROVED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (george-silva) [reply .github/workflows/release.yml]: Ok, sure.

**Comment by logrock-eng** [comment]
# Netlify Preview Deployed 

 Preview: https://68--logrock-pebbles.netlify.app 

 ## View on Browserstack 

 https://live.browserstack.com/dashboard#os=iOS&os_version=15.0&device_browser=safari&zoom_to_fit=true&full_screen=true&url=https://68--logrock-pebbles.netlify.app&speed=1

**Pull request merged**